TheSadRogue.Integration/RogueLikeEntity.cs

```diff
@@ -131,18 +109,14 @@ public T GetComponent<T>(string tag = "")
             //     return GetComponents<T>().Distinct().FirstOrDefault();
             // }
         }
-        public IEnumerable<IRogueLikeComponent> GetComponents()
-            => GoRogueComponents.GetAll<IRogueLikeComponent>().Concat(GetSadComponents<IRogueLikeComponent>());
+        public IEnumerable<object> GetComponents()
+            => GoRogueComponents.GetAll<object>().Concat(GetSadComponents<IComponent>()).Distinct();
 
         public IEnumerable<T> GetComponents<T>()
```

Comment:
I can't apply a suggestion over the implementation below due to deleted lines; however this function should really just be implemented as: 
```suggestion
        public IEnumerable<T> GetComponents<T>() => GoRogueComponents.GetAll<T>();
```

As stated elsewhere, the way `AddComponents` is implemented `GoRogueComponents` contains _all_ components added via `AddComponent`; and if you're maintaining the lists separately these functions are of very little use to you anyway (as their context doesn't make a lot of sense).  Implementing a try-cast here is less asymptotically efficient than the operation that `GetAll` provides.  Further, as mentioned on the `GetComponents` function, that function should have no need to do a `Distinct` call as `GoRogueComponents` contains all the joint components.

SadConsole's component structure is very heavily optimized for iteration (which is exactly what SadConsole uses it for on a frame-by-frame basis); but for general (non-iteration) uses, IMO GoRogue functions provide better asymptotic scaling; so when given the choice, I would generally prefer GoRogue's.

Reply:
Sounds reasonable. Unfortunately, the `GetAll<T>` solution isn't working because `T must be a reference type`. Fix incoming.

Reply:
Yes, components that are not reference types should be disallowed; GoRogue can't handle them.